app/server/lib/OIDCConfig.ts

```diff
@@ -203,30 +249,93 @@ export class OIDCConfig {
       return this._endSessionEndpoint;
     }
     return this._client.endSessionUrl({
-      post_logout_redirect_uri: redirectUrl.href
+      post_logout_redirect_uri: redirectUrl.href,
+      state: mreq.session.oidc?.state,
+      id_token_hint: mreq.session.oidc?.idToken,
+    });
```

Comment:
I don't clear either the `state` or the `idToken` in this method, as I feel like it's more secure to secure the logout if the first attempt fails (the following attempts may succeed).

Reply:
Here, you can see that the whole `mreq.session.oidc` object used to be clear entirely and now they continue containing the `state` and the `idToken` properties (these info will be cleaned up with the whole session deletion):

https://github.com/gristlabs/grist-core/pull/883/files#diff-ee4d5464063d6cfc9dc24b7b63d8858ff7b978dbe767ad2a712c93fe52f7ca1dR224

Reply:
Sounds good, thanks!